db/db.go

```diff
@@ -5610,6 +5645,83 @@ func DeleteDlsMsgsByTenant(tenantName string) error {
 
 }
 
+func GetMinMaxIdDlsMsgsByUpdatedAt(tenantName string, createdAt time.Time, stationId int) (bool, []models.DlsMsgResendAll, int, int, error) {
+	ctx, cancelfunc := context.WithTimeout(context.Background(), DbOperationTimeout*time.Second)
+	defer cancelfunc()
+	conn, err := MetadataDbClient.Client.Acquire(ctx)
+	if err != nil {
+		return false, []models.DlsMsgResendAll{}, -1, -1, err
+	}
+	defer conn.Release()
+	query := `
+		WITH limited_rows AS (
+		SELECT *
+		FROM dls_messages
+		WHERE tenant_name = $1 AND updated_at <= $2 AND message_type = 'poison' AND station_id = $3
+		)
+		SELECT *,
+		(SELECT MIN(id) FROM limited_rows) AS min_id,
+		(SELECT MAX(id) FROM limited_rows) AS max_id
+		FROM limited_rows;`
+	stmt, err := conn.Conn().Prepare(ctx, "get_min_max_id_dls_msgs_by_updated_at", query)
+	if err != nil {
+		return false, []models.DlsMsgResendAll{}, -1, -1, err
+	}
+	if tenantName != conf.GlobalAccount {
+		tenantName = strings.ToLower(tenantName)
+	}
+	rows, err := conn.Conn().Query(ctx, stmt.Name, tenantName, createdAt, stationId)
+	if err != nil {
+		return false, []models.DlsMsgResendAll{}, -1, -1, err
+	}
+	defer rows.Close()
+
+	dlsMsgs, err := pgx.CollectRows(rows, pgx.RowToStructByPos[models.DlsMsgResendAll])
+	if err != nil {
+		return false, []models.DlsMsgResendAll{}, -1, -1, err
+	}
+	if len(dlsMsgs) == 0 {
+		return false, []models.DlsMsgResendAll{}, -1, -1, nil
+	}
+	return true, dlsMsgs, dlsMsgs[0].MinId, dlsMsgs[0].MaxId, nil
+}
+
+func GetDlsMsgsBatch(tenantName string, min, max, stationId int) (bool, []models.DlsMessage, error) {
+	ctx, cancelfunc := context.WithTimeout(context.Background(), DbOperationTimeout*time.Second)
+	defer cancelfunc()
+	conn, err := MetadataDbClient.Client.Acquire(ctx)
+	if err != nil {
+		return false, []models.DlsMessage{}, err
+	}
+	defer conn.Release()
+	query := `
+		SELECT *
+		FROM dls_messages
+		WHERE tenant_name = $1 AND message_type = 'poison' AND station_id = $2 AND id > $3 AND id <= $4 ORDER BY id ASC LIMIT 10
```

Comment:
When >= this will include in the next batch the last message as the first message and will actually fetch the same message once more according to the offset